### PR TITLE
KAFKA-12318: system tests need to fetch Topic IDs via Admin Client instead of via ZooKeeper

### PR DESCRIPTION
**Pull request by jolshan**
Changes system tests to support both ZK and raft topic IDs. Clarifies the IBP check is for zk code.

Ran `upgrade_test.py` and `downgrade_test.py` as these both use the `all_nodes_support_topic_ids` and/or `topic_id` methods. Currently does not include tests using raft, but verified that the output for describe topics in raft code returns same  output format and contains a valid topic ID.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tests/kafkatest/services/kafka/kafka.py [hunk 1]
```diff
@@ -1120,7 +1121,7 @@ def run_cli_tool(self, node, cmd):
 
     def parse_describe_topic(self, topic_description):
         """Parse output of kafka-topics.sh --describe (or describe_topic() method above), which is a string of form
-        PartitionCount:2\tReplicationFactor:2\tConfigs:
+        Topic:test_topic\tTopicId:<topic_id>\tPartitionCount:2\tReplicationFactor:2\tConfigs:
```
Comment (rondagostino) [inline tests/kafkatest/services/kafka/kafka.py]: Add the spaces?  (e.g. `Topic: test_topic\tTopicId: <topic_id>\tPartitionCount: 2\tReplicationFactor: 2\tConfigs:`

Also, in next line, `s/tPartition/ Partition/` I think.

File: tests/kafkatest/version.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def kafka_configs_command_uses_bootstrap_server_scram(self):
         # User SCRAM Credentials (KIP-554)
         return self >= V_2_7_0
 
-    def supports_topic_ids(self):
+    def zk_supports_topic_ids(self):
```
Comment (rondagostino) [inline tests/kafkatest/version.py]: Maybe `supports_topic_ids_when_using_zk` is a better name since it isn't ZooKeeper that is supporting topic IDs?

**Review by rondagostino — COMMENTED** [review]
Thanks for the PR -- LGTM!  Just a couple of minor things.

File: tests/kafkatest/version.py (hunk 2 above)
Comment (jolshan) [reply tests/kafkatest/version.py]: I was having trouble naming this haha. This is a better name.

File: tests/kafkatest/services/kafka/kafka.py (hunk 1 above)
Comment (jolshan) [reply tests/kafkatest/services/kafka/kafka.py]: Makes sense.

**Comment by jolshan** [comment]
System tests `upgrade_test.py` and `downgrade_test.py` passed with latest code.

**Comment by cmccabe** [comment]
Hmm, I'm not sure why we want to rename `supports_topic_ids` to `supports_topic_ids_when_using_zk`.  KIP-500 mode is only supported from 2.8 on, and does fully support topic ids.

**Comment by jolshan** [comment]
@cmccabe there's a small distinction in that kip-500 can run on ibp < 2.8, where zk IDs are created only when ibp is 2.8 or higher. 

My understanding is that if we are using kip-500 mode, then we have topic IDs. This is the logic behind `all_nodes_support_topic_ids`

**Comment by cmccabe** [comment]
KIP-500 doesn't really run on IBP 2.7 or earlier since AlterIsr would be de-activated.

Well, technically you could run 2.7. internal version 2 to get it, but we don't test internal versions in ducktape (they don't appear in `version.py`) and I don't think they're much used in production either...

**Comment by jolshan** [comment]
@cmccabe I think ducktape runs on the latest IBP for the release. So it would run 2.7-IV2

**Comment by jolshan** [comment]
I think ducktape uses `ApiVersion` and `assertEquals(KAFKA_2_7_IV2, ApiVersion("2.7"))` succeeds.

**Comment by cmccabe** [comment]
That's a fair point.

Can you add a comment to `supports_topic_ids_when_using_zk` explaining that self-managed clusters always support topic IDs?

LGTM after that.

**Review by cmccabe — APPROVED** [review]

**Pull request merged**